moveit_ros/planning/trajectory_execution_manager/cfg/TrajectoryExecutionDynamicReconfigure.cfg

```diff
@@ -7,5 +7,6 @@ gen.add("execution_duration_monitoring", bool_t, 1, "Monitor the execution durat
 gen.add("allowed_execution_duration_scaling", double_t, 2, "Accept durations that take a little more time than specified", 1.1, 1, 10)
 gen.add("allowed_goal_duration_margin", double_t, 3, "Allow more than the expected execution time before triggering a trajectory cancel (applied after scaling)", 0.5, 0.1, 5)
 gen.add("execution_velocity_scaling", double_t, 4, "Multiplicative factor for execution speed", 1, 0.1, 10)
+gen.add("allowed_start_tolerance", double_t, 5, "Allowed joint-value tolerance for validation of trajectory's start point against current robot state", 1e-3, 0);
```

Comment:
I think I would lean toward leaving this out of dyn reconfig. Your bullets sum up the arguments I would've used as well.

Beyond this one PR though, I can see making things like this dynamically (re)configurable being useful when requirements change during runtime. Two examples:

 1. I know of people using MoveIt with drones and other more mobile robots. Perhaps even just to add some 'temporary' mobility to an otherwise static manipulator, but still. I would expect -- but cannot back it up with nrs right now -- that noise in joint states would increase during motion. Manipulation during motion would then need to have (a slightly more) relaxed threshold(s). Afterwards, the old setting could be restored.

 1. (slightly more of a stretch, as considering MoveIt as part of the safety features of any robot system is currently not even considerable): in the context of co-robots, changing thresholds / safety related distances and things like maximum velocities and forces based on the proximity of humans becomes more and more common place.

  I don't see MoveIt used for these things (in production systems at least), but dynamically changeable parameters would seem to be a requirement for this kind of functionality.